src/tbb/arena.cpp

```diff
@@ -254,15 +324,12 @@ void arena::free_arena () {
 #if __TBB_PREVIEW_CRITICAL_TASKS
     __TBB_ASSERT( my_critical_task_stream.empty(), "Not all critical tasks were executed");
 #endif
-    // remove an internal reference
-    my_market->release( /*is_public=*/false, /*blocking_terminate=*/false );
-
     // Clear enfources synchronization with observe(false)
     my_observers.clear();
 
     void* storage  = &mailbox(my_num_slots-1);
     __TBB_ASSERT( my_references.load(std::memory_order_relaxed) == 0, nullptr);
-    __TBB_ASSERT( my_pool_state.load(std::memory_order_relaxed) == SNAPSHOT_EMPTY || !my_max_num_workers, nullptr);
+    __TBB_ASSERT( my_pool_state.test() == false|| !my_max_num_workers, nullptr);
```

Comment:
Debug vs Release memory semantics discrepancy detected. Perhaps, add `test_debug()` method under the debug `ifdef` if the assert is that useful?

Reply:
Agreed on adding memory semantics argument to test

Reply:
Done